### PR TITLE
feat(data_structures): improve remove method in TransactionsPool to r…

### PR DESCRIPTION
**Pull request by lrubiorod**
…emove all the transactiosn with the same input

### CODE REVIEW

File: data_structures/src/chain.rs [hunk 1]
```diff
@@ -1671,6 +1674,24 @@ impl TransactionsPool {
...
+                    }
+                }
+            }
+        }
+
+        transaction
+    }
+
+    /// Value Transfer remove inner
+    fn vt_remove_inner(&mut self, key: &Hash) -> Option<VTTransaction> {
```
Comment (aesedepece) [inline data_structures/src/chain.rs:1694]: Quite opaque of a function name and comment :rofl: 
```suggestion
    /// Remove a value transfer transaction from the pool without any further guarantee about removing other transactions that may try to spend the same UTXOs.
    fn vt_remove_inner(&mut self, key: &Hash) -> Option<VTTransaction> {
```

File: data_structures/src/chain.rs [hunk 2]
```diff
@@ -1671,6 +1674,24 @@ impl TransactionsPool {
     /// assert!(!pool.vt_contains(&transaction.hash()));
     /// ```
     pub fn vt_remove(&mut self, key: &Hash) -> Option<VTTransaction> {
```
Comment (aesedepece) [inline data_structures/src/chain.rs:1676]:
```suggestion
    /// Remove a value transfer transaction from the pool and make sure that other transactions that may try to spend the same UTXOs are also removed.
    pub fn vt_remove(&mut self, key: &Hash) -> Option<VTTransaction> {
```

File: data_structures/src/chain.rs [hunk 3]
```diff
@@ -1701,6 +1722,24 @@ impl TransactionsPool {
     /// assert!(!pool.dr_contains(&transaction.hash()));
     /// ```
     pub fn dr_remove(&mut self, key: &Hash) -> Option<DRTransaction> {
```
Comment (aesedepece) [inline data_structures/src/chain.rs:1724]: /// Remove a data request transaction from the pool and make sure that other transactions that may try to spend the same UTXOs are also removed.

File: data_structures/src/chain.rs [hunk 4]
```diff
@@ -1701,6 +1722,24 @@ impl TransactionsPool {
...
+                    }
+                }
+            }
+        }
+
+        transaction
+    }
+
+    /// Data Request remove inner
+    fn dr_remove_inner(&mut self, key: &Hash) -> Option<DRTransaction> {
```
Comment (aesedepece) [inline data_structures/src/chain.rs:1742]:
```suggestion
    /// Remove a data request transaction from the pool without any further guarantee about removing other transactions that may try to spend the same UTXOs.
    fn dr_remove_inner(&mut self, key: &Hash) -> Option<DRTransaction> {
```

**Comment by tmpolaczyk** [comment]
As mentioned in #1308, this will increase the memory usage quite a bit.

We would need to test this in a real world scenario to see if it is worth it.

**Comment by lrubiorod** [comment]
> As mentioned in #1308, this will increase the memory usage quite a bit.
> 
> We would need to test this in a real world scenario to see if it is worth it.

Totally agree. Moreover, unless we are increasing the memory usage quite a bit, when we combine this solution with the possibility to validate the transactions once, and not every time that we see in a block, that will implies less validations and less memory usage

**Comment by aesedepece** [comment]
Memory pools can be limited in size anyway, as they are only a nice-to-have for mining and speeding up validations, but not strictly needed for keeping a node up to date.

**Pull request closed without merging**